lib/token-translator.js

```diff
@@ -123,8 +123,9 @@ TokenTranslator.prototype = {
                  type === tt.arrow || type === tt.jsxTagStart ||
                  type === tt.incDec || type === tt.starstar ||
                  type === tt.jsxTagEnd || (type.binop && !type.keyword) ||
-                 type.isAssign) {
-
+                 type.isAssign ||
+                 (type.prefix && (token.value === "~" || token.value === "!"))
```

Comment:
Do we need to filter further if `type.prefix` is true? Is there ever a situation where a prefix isn't a punctuation? Or are there other punctuator prefixes that this should also cover?